tests/lib/rules/no-get-by-for-asserting-element-not-present.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/no-get-by-for-asserting-element-not-present');
+const { ALL_QUERIES_METHODS } = require('../../../lib/utils');
+
+const ruleTester = new RuleTester({
+  parserOptions: { ecmaVersion: 2017, sourceType: 'module' },
+});
+
+const getByVariants = ALL_QUERIES_METHODS.reduce(
+  (variants, method) => [...variants, ...[`get${method}`, `getAll${method}`]],
+  []
+);
+
+ruleTester.run('prefer-expect-query-by', rule, {
+  valid: getByVariants.reduce(
+    (validRules, queryName) => [
+      ...validRules,
+      { code: `expect(${queryName}('Hello')).toBeInTheDocument()` },
+      { code: `expect(${queryName}('Hello')).toBe("foo")` },
+      { code: `expect(${queryName}('Hello')).toBeTruthy()` },
+      { code: `expect(rendered.${queryName}('Hello')).toEqual("World")` },
+      { code: `expect(rendered.${queryName}('Hello')).not.toBeFalsy()` },
+    ],
```

Comment:
We need valid tests for `waitForElementToBeRemoved`, which is the second part of the rule.